packages/edit-post/src/components/sidebar/template/actions.js

```diff
@@ -79,20 +84,50 @@ function PostTemplateActions() {
 						onSubmit={ ( event ) => {
 							event.preventDefault();
 							const defaultTitle = __( 'Custom Template' );
-							const templateContent = [
-								createBlock( 'core/site-title' ),
-								createBlock( 'core/site-tagline' ),
-								createBlock( 'core/separator' ),
-								createBlock( 'core/post-title' ),
-								createBlock( 'core/post-content', {
-									layout: { inherit: true },
-								} ),
-							];
+							const newTemplateContent =
+								defaultTemplate ??
+								serialize( [
+									createBlock(
+										'core/group',
+										{
+											tagName: 'header',
+											layout: { inherit: true },
+										},
+										[
+											createBlock( 'core/site-title' ),
+											createBlock( 'core/site-tagline' ),
+										]
+									),
+									createBlock( 'core/separator' ),
+									createBlock(
+										'core/group',
+										{
+											tagName: 'main',
+										},
+										[
+											createBlock(
+												'core/group',
+												{
+													layout: { inherit: true },
+												},
+												[
+													createBlock(
+														'core/post-title'
+													),
+												]
+											),
+											createBlock( 'core/post-content', {
+												layout: { inherit: true },
+											} ),
+										]
+									),
```

Comment:
Hmmm everything looks perfect, with the exception of the post-title block which is nested inside a 2nd group block for some reason :thinking: `Post → Group → Group → Post Title`
Was that intentional or should it be like this instead?

```suggestion
									createBlock(
										'core/group',
										{
											tagName: 'main',
											layout: { inherit: true },
										},
										[
											createBlock( 'core/post-title' ),
											createBlock( 'core/post-content', {
												layout: { inherit: true },
											} ),
										]
									),
```

Reply:
> Was that intentional or should it be like this instead?

yes, it's intentional, that way the wrapper serves as a "main" but it doesn't define a layout, so the post title will be wide by default, the second group defines a layout making the title block centered by default like the default content in the post content.

Reply:
Ah ok, gotcha. Makes sense :+1: